drivers/dma/dma_mcux_edma.c

```diff
@@ -205,24 +247,56 @@ static int dma_mcux_edma_configure(const struct device *dev, uint32_t channel,
 		return -EINVAL;
 	}
 
-	data->dir = config->channel_direction;
+	data->transfer_settings.valid = false;
+
 	switch (config->channel_direction) {
-	case MEMORY_TO_MEMORY:
+	case MEMORY_TO_MEMORY: {
```

Comment:
These brackets in this switch case aren't necessary

Reply:
I thought uncrustify added these for me, but I will remove for now.

Reply:
We are moving away from uncrustify, I know the docs still mention them, but please do try the newer clang format setup if you are interested in doing auto formatting like me.  See #41307